### ISSUE TITLE
Architecture revision and redesign

### ISSUE THREAD

**Issue by DmitrySikorsky** [Platformus/Platformus#210]
I'd started working on the first version of the Platformus CMS on 2015. Currently I have used it in many projects. I really like some ideas, but based on experience, much needs to be changed. Now I understand better how the system should be designed. I want to make it lightweight and faster, more flexible etc. It will be much, much easier than now.

**Comment by weedkiller** [comment]
@DmitrySikorsky you should also consider using some of these libs to help speed up platforms.  
[Awesome Dot Net](https://github.com/optimajet/awesome-dotnet)

**Comment by MosesMachua** [comment]
Hi Dmitry. I've been following this project for a while as well as ExtCore and I've got to say I'm quite impressed. Kudos for a job well done and I can't wait to see what enhancements you have in store. 

One of my suggestions for enhancements is to have a `built-in installer for extensions`. I've used [DotNetNuke (DNN)](https://github.com/dnnsoftware/Dnn.Platform) for years and this is one of its greatest features - the ability to distribute extensions as zip files to my clients that they can themselves install through the built in installer. DNN being ASP.NET webforms based uses the [web.config trick](https://forums.asp.net/t/1569137.aspx?Force+an+ASP+net+Application+to+recompile+itself+during+runtime+) to make the application restart so the new extension assemblies can be loaded. Does Platformus / ExtCore have a built in restart method or a way to discover newly added extensions at runtime?

Keep up the good work. Thanks
